scripts/mh_windowfill.pl

```diff
@@ -0,0 +1,244 @@
+
+##############################################################################
+#
+# common support functions
+#
+##############################################################################
+
+sub trim_spacelike($)
+{
+   my ($string) = @_;
+
+   if (defined($string))
+   {
+      $string =~ s/^\s+//g;
+      $string =~ s/\s+$//g;
+
+   } else {
+
+      $string = '';
+   }
+
+   return($string);
+}
+
+##############################################################################
+#
+# script functions
+#
+##############################################################################
+
+sub windowfill($)
+{
+	my ($window) = @_;
+
+	if (ref($window) ne 'Irssi::UI::Window')
+	{
+		die();
+	}
+
+	#
+	# fill window with empty lines and move already printed lines to the bottom
+	#
+	if (($window->view()->{'ypos'} + 2) <= $window->{'height'})
+	{
+
+		while (($window->view()->{'ypos'} + 2) <= $window->{'height'})
+		{
+			$window->print('', MSGLEVEL_CLIENTCRAP | MSGLEVEL_NEVER | MSGLEVEL_NO_ACT | MSGLEVEL_NOHILIGHT);
+		}
+
+		my $linecount  = $window->{'height'};
+		my $line       = $window->view()->get_lines();
+
+		while ((ref($line) eq 'Irssi::TextUI::Line') and $linecount)
+		{
+			my $linetext = $line->get_text(1);
+
+			if ($linetext ne '')
+			{
+				# reprint line
+				$window->print($linetext, MSGLEVEL_CLIENTCRAP | MSGLEVEL_NEVER | MSGLEVEL_NO_ACT | MSGLEVEL_NOHILIGHT);
+				$line = $line->next();
+				$window->view()->remove_line($line->prev());
+
+			} else {
+
+				# skip empty line
+				$line = $line->next();
+			}
+
+			$linecount--;
+		}
+	}
+
+	return(1);
+}
+
+sub windowfill_all()
+{
+	#
+	# fill all windows with empty lines
+	#
+	for my $window (Irssi::windows())
+	{
+		if (ref($window) ne 'Irssi::UI::Window')
+		{
+			die();
+		}
+
+		windowfill($window);
+	}
+
+	return(1);
+}
+
+##############################################################################
+#
+# irssi signal handlers
+#
+##############################################################################
+
+sub signal_mainwindow_resized_last()
+{
+	if ($windowfill_running)
+	{
+		#
+		# fill all windows with empty lines
+		#
+		windowfill_all();
+	}
+
+	return(1);
+}
+
+Irssi::signal_add_last('mainwindow resized', 'signal_mainwindow_resized_last');
+
+sub signal_window_created_last($)
+{
+	my ($window) = @_;
+
+	if (ref($window) ne 'Irssi::UI::Window')
+	{
+		die();
+	}
+
+	if ($windowfill_running)
+	{
+		#
+		# fill created window with empty lines
+		#
+		windowfill($window);
+	}
+
+	return(1);
+}
+
+Irssi::signal_add_last('window created', 'signal_window_created_last');
+
+##############################################################################
+#
+# script on load
+#
+##############################################################################
+
+sub script_on_load($)
+{
+	my ($undef) = @_;
+
+	if (defined($undef))
+	{
+		die();
+	}
+
+	if ($windowfill_running)
+	{
+		die();
+	}
+
+	windowfill_all();
+	$windowfill_running = 1;
```

Comment:
I keep my souce seperated by function and use preprocessing to create the signal_add in the files where i create the signal_ subs, this might be something to fix in the future, but for now it works